plugins/jobs/src/js/components/form/GeneralFormSection.tsx

```diff
@@ -0,0 +1,365 @@
+  return errors
+    .filter(e => {
+      const match = e.path.join(".");
+      return match === path;
+    })
+    .map(e => e.message)
+    .join(" ");
+}
+
+class GeneralFormSection extends Component<GeneralProps, object> {
+  constructor(props: GeneralProps) {
+    super(props);
+  }
+
+  getResourceRow() {
+    const {
+      formData: { job },
+      showErrors,
+      errors
+    } = this.props;
+    const cpuTooltipContent =
+      "The number of CPU shares this job needs per instance. This number does not have to be integer, but can be a fraction.";
+    const diskTooltipContent =
+      "How much disk space is needed for this job. This number does not have to be an integer, but can be a fraction.";
+    const memTooltipContent =
+      "The amount of memory in MB that is needed for the job per instance.";
+    const gpuTooltipContent =
+      "The number of GPU shares this job needs per instance. This number does not have to be integer, but can be a fraction.";
+    const gpusDisabled = !job.run.ucr;
+
+    const cpusError = getFieldError("run.cpus", errors);
+
+    return (
+      <FormRow>
+        <FormGroup className="column-3" showError={showErrors && cpusError}>
+          <FieldLabel>
+            <FormGroupHeading required={true}>
+              <FormGroupHeadingContent title="CPUs" primary={true}>
+                <Trans render="span">CPUs</Trans>
+              </FormGroupHeadingContent>
+              <FormGroupHeadingContent title="CPUs Info">
+                <Tooltip
+                  content={cpuTooltipContent}
+                  interactive={true}
+                  maxWidth={300}
+                  wrapText={true}
+                >
+                  <Icon color="light-grey" id="circle-question" size="mini" />
+                </Tooltip>
+              </FormGroupHeadingContent>
+            </FormGroupHeading>
+          </FieldLabel>
+          <FieldInput min="0.01" name="job.run.cpus" value={job.run.cpus} />
+          <FieldError>{cpusError}</FieldError>
+        </FormGroup>
+
+        <FormGroup className="column-3">
+          <FieldLabel className="text-no-transform">
+            <FormGroupHeading required={true}>
+              <FormGroupHeadingContent title="Mem">
+                <Trans render="span">Mem (MiB)</Trans>
+              </FormGroupHeadingContent>
+              <FormGroupHeadingContent title="Mem Info">
+                <Tooltip
+                  content={memTooltipContent}
+                  interactive={true}
+                  maxWidth={300}
+                  wrapText={true}
+                >
+                  <Icon color="light-grey" id="circle-question" size="mini" />
+                </Tooltip>
+              </FormGroupHeadingContent>
+            </FormGroupHeading>
+          </FieldLabel>
+          <FieldInput
+            min="32"
+            name="job.run.mem"
+            type="number"
+            value={job.run.mem}
```

Comment:
Something we should do for all nested fields is we should use [findNestedPropertyInObject](https://github.com/dcos/dcos-ui/blob/master/src/js/utils/Util.js#L79-L91) We can't be sure that the fields are present end if `run` is not present the whole form will break.